db/keyval/etcd/config.go

```diff
@@ -40,6 +40,7 @@ type Config struct {
 	CAfile                string        `json:"ca-file"`
 	AutoCompact           time.Duration `json:"auto-compact"`
 	ReconnectResync       bool          `json:"resync-after-reconnect"`
+	DelayedStart          bool          `json:"delayed-start"`
```

Comment:
`DelayedRestart` sounds like we want the start to be delayed, even though the option actually means we want to allow delayed start/connect. 

I'd suggest to use either:
 - `DialRetry` which would also be more fitting with `DialTimeout` option we currently have
 - or at least put `AllowDelayedStart` to make it clear that we are allowing start to be delayed

@lukasmacko what do you think?

Reply:
I'm for AllowDelayedStart, that was the original name of the field, but it seemed too long for me